cirq/linalg/decompositions.py

```diff
@@ -514,6 +528,120 @@ def interaction_matrix(m: np.ndarray, c: float) -> np.ndarray:
+            if x = 1 then z >= 0
+
+        The wireframe includes lines along the surface of the space at z=0.
+
+    Args:
+        interactions: An iterable of two qubit unitary interactions. Each
+            interaction can be specified as a raw 4x4 unitary matrix, or an
+            object with a 4x4 unitary matrix according to `cirq.unitary` (
+            (e.g. `cirq.CZ` or a `cirq.KakDecomposition` or a `cirq.Circuit`
+            over two qubits).
+        include_frame: Determines whether or not to draw the kak space
+            wireframe. If not set. Defaults to `True`.
+        ax: A matplotlib 3d axes object to plot into. If not specified, a new
+            figure is created.
+        show: Whether or not to call `matplotlib.pyplot.show()`. Defaults to
+            `False`.
+        kwargs: Arguments forwarded into the call to `scatter` that plots the
+            points. Working arguments include color `c='blue'`, scale `s=2`,
+            labelling `label="theta=pi/4"`, etc. For reference see the
+            `matplotlib.pyplot.scatter` documentation:
+            https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.scatter.html
+
+    Returns:
+        The matplotlib 3d axes object that was plotted into.
+
+    Examples:
+        >>> ax = None
+        >>> for y in np.linspace(0, 0.5, 4):
+        ...     a, b = cirq.LineQubit.range(2)
+        ...     circuits = [
+        ...         cirq.Circuit.from_ops(
+        ...             cirq.CZ(a, b)**0.5,
+        ...             cirq.X(a)**y, cirq.X(b)**x,
+        ...             cirq.CZ(a, b)**0.5,
+        ...             cirq.X(a)**x, cirq.X(b)**y,
+        ...             cirq.CZ(a, b) ** 0.5,
+        ...         )
+        ...         for x in np.linspace(0, 1, 25)
+        ...     ]
+        ...     ax = cirq.scatter_plot_normalized_kak_interaction_coefficients(
+        ...         circuits,
+        ...         s=1,
+        ...         ax=ax,
+        ...         include_frame=ax is None,
+        ...         label=f'y={y:0.2f}')
+        >>> _ = ax.legend()
+        >>> import matplotlib.pyplot as plt
+        >>> plt.show()
+    """
+    if ax is None:
+        fig = plt.figure()
+        ax = fig.add_subplot(projection='3d')
+
+    def coord_transform(
+            pts: Iterable[Tuple[float, float, float]]
+    ) -> Tuple[Iterable[float], Iterable[float], Iterable[float]]:
+        xs, ys, zs = zip(*pts)
+        return xs, zs, ys
+
+    if include_frame:
+        envelope = [
+            (0, 0, 0),
+            (1, 1, 1),
+            (1, 1, -1),
+            (0, 0, 0),
+            (1, 1, 1),
+            (1, 0, 0),
+            (0, 0, 0),
+            (1, 1, -1),
+            (1, 0, 0),
+            (0, 0, 0),
+            (1, 0, 0),
+            (1, 1, 0),
+            (0, 0, 0),
+        ]
+        ax.plot(*coord_transform(envelope), c='black', linewidth=1)
+
+    points = []
+    for obj in interactions:
+        kak = kak_decomposition(obj)
```

Comment:
I haven't found time to read the reference paper on KAK decomposition, but I take it that it's some logic that transforms the object into a KakDecomposition whose `interaction_coefficients` are to be plotted in a 3D axes. That sounds like mixing business logic (the transformation) with presentation (the plotting).

How about making this a method of the KakDecomposition class?

Reply:
I think this is fine as-is. AFAIU, the purpose of `scatter_plot_normalized_kak_interaction_coefficients()`  is to help in interactive work where one repeatedly examines the coefficients of different variations of a unitary. Splitting out the decomposition and presentation would force the user to call two functions instead of one.

Also, this particular presentation is tailored to the specifics of the decomposition.

Reply:
> That sounds like mixing business logic (the transformation) with presentation (the plotting).

We already have support for getting the interaction coefficients and matplotlib already has methods for making scatter plots. This method puts the two together in a convenient way. Factoring it into two methods would produce no changes to the code base, because the two pieces already exist.

Reply:
> How about making this a method of the KakDecomposition class?

I originally did, but it had to be a static method (because we're talking about a *list* of values instead of a single value) and also this made it less clear that you could include things that were not predecomposed in the list.